include/aikido/control/TrajectoryExecutor.hpp

```diff
@@ -20,7 +20,7 @@ class TrajectoryExecutor
   /// Validate the traj in preparation for execution.
   ///
   /// \param traj Trajectory to be validated
-  virtual void validate(trajectory::TrajectoryPtr traj) = 0;
+  virtual void validate(const trajectory::ConstTrajectoryPtr& traj) = 0;
```

Comment:
Does `validate` change the internal states of a trajectory executor? If not, let's make this a const function.